src/Popover/modifiers.js

```diff
@@ -11,35 +12,47 @@ const computeArrowPadding = () => {
     return Math.ceil(padding)
 }
 
-export const offset = {
-    name: 'offset',
-    options: {
-        offset: [0, ARROW_SIZE],
-    },
-}
+const hideArrowWhenDisplaced = ({ state }) => {
+    const halfArrow = ARROW_SIZE / 2
+    const displacement = state.modifiersData.preventOverflow
+    const referenceRect = state.rects.reference
+    const shouldHideArrow =
+        Math.abs(displacement.x) >= referenceRect.width + halfArrow ||
+        Math.abs(displacement.y) >= referenceRect.height + halfArrow
+
+    state.attributes.arrow = { ['data-arrow-hidden']: shouldHideArrow }
```

Comment:
Should this be `state.attributes.arrow = { ...state.attributes.arrow, ['data-arrow-hidden']: shouldHideArrow }` or will there never be any other attributes?

Reply:
Ah I see we're just using this mechanism ourselves, so it should always be only that property.

Reply:
Yes, but still a very valid point. I've fixed this in https://github.com/dhis2/ui-core/pull/886/commits/ac0529302c54c51fb7c8c9274cf50534aa06baf1